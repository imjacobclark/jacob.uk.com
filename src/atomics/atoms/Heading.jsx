import React from 'react';

export default class Heading extends React.Component {
    render(){
        const Level = this.props.level;
        return <Level>{this.props.children}</Level>
    }
}