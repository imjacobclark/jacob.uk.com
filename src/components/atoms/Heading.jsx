import React from 'react';

export default ({level, children}) => {
    const Heading = `h${level}`;
    return <Heading>{children}</Heading>
}