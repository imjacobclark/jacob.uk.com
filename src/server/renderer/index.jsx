import App from '../../components/App.jsx';

export default dependencies => () => {
    const React = dependencies.getReact();
    const ReactDOMServer = dependencies.getReactDOMServer();

    return `
<html>
    <head>
    </head>
    <body>
        <main id="jacobclarkxyz__main">
            ${ReactDOMServer.renderToString(<App runtime="server"/>)}
        </main>
        <script src="client.bundle.js"></script>
    </body>
</html>`;
}