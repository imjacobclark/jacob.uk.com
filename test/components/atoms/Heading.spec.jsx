import React from 'react';
import renderer from 'react-test-renderer';
import Heading from '../../../src/components/atoms/Heading.jsx';

describe('Heading', () => {
    it('renders a h1 correctly', () => {
        const tree = renderer.create(<Heading level="1">Jacob Clark</Heading>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders a h2 correctly', () => {
        const tree = renderer.create(<Heading level="2">Jacob Clark</Heading>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
