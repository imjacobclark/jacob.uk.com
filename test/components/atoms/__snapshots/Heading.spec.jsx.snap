// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server routes renders correctly 1`] = `
<h1>
  Jacob Clark
</h1>
`;